docs/articles/documentation/extending-testcafe/creating-custom-reporter/index.md

```diff
@@ -0,0 +1,143 @@
+---
+layout: docs
+title: Creating Custom Reporter
+permalink: /documentation/extending-testcafe/creating-custom-reporter/
+---
+# Creating Custom Reporter
+
+TestCafe has a number of [built-in reporters](/testcafe/documentation/using-testcafe/common-concepts/reporters/) to generate test reports in different formats.
+You can also create a **custom reporter** that will output information in your own format and style.
+For this purpose, you can use the [TestCafe reporter generator](https://github.com/DevExpress/generator-testcafe-reporter).
+The generator will scaffold out a reporter plugin, so that you only need to write a few lines of code.
+
+## Generating Reporter Project
+
+First, install [Yeoman](http://yeoman.io) and `generator-testcafe-reporter` using [npm](https://www.npmjs.com/) (we assume you have pre-installed [node.js](https://nodejs.org/)).
+
+```bash
+npm install -g yo
+npm install -g generator-testcafe-reporter
+```
+
+Create a new directory and go into it.
+
+```bash
+mkdir MyReporter
+cd MyReporter
```

Comment:
It's important. The name of the directory is usually the name of the project (e.g. it will be strange to have testcafe source in `MyProject` dir, we have a `testcafe` dir instead).  And generator's prompt tries to obtain suggested project name from the dir name.